state/core_access.go

```diff
@@ -501,13 +543,53 @@ func (ca *CoreAccessor) PayForBlobCount() int64 {
 	return ca.payForBlobCount
 }
 
+// SetMetrics updates the metrics for the CoreAccessor
+func (ca *CoreAccessor) SetMetrics(metrics *metrics) {
```

Comment:
Lets remove this from public api as its cant and not going to be used. I think we can directly assign to metrics field in place without the method and locking.

Reply:
done